.gitlab/scripts.yml

```
@@ -29,8 +29,8 @@
       export OMPI_ALLOW_RUN_AS_ROOT=1;
       export OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1;
       fi
-    - if [[ "${GINKGO_BUILD_MPI}" == "ON" ]]; then
-      MPI_STR=-DGINKGO_MPI_EXEC_SUFFIX=${MPI_SUFFIX} -DMPI_RUN_AS_ROOT=${MPI_AS_ROOT};
+    - if [[ "${BUILD_MPI}" == "ON" ]]; then
+      MPI_STR=-DGINKGO_MPI_EXEC_SUFFIX=${MPI_SUFFIX};
```

```suggestion
      MPI_STR=-DGINKGO_MPI_EXEC_SUFFIX=${MPI_SUFFIX} -DMPI_RUN_AS_ROOT=${MPI_AS_ROOT};
```
create_test.cmake needs `MPI_RUN_AS_ROOT` to add `--allow-run-as-root`

---

No, I think either the flag is needed or the environment variable needs to be set, which we do. See https://www.open-mpi.org/doc/v4.0/man1/mpirun.1.php#sect25

---

In that case, this block as well as the logic here can be dropped entirely, no?
https://github.com/ginkgo-project/ginkgo/blob/develop/cmake/create_test.cmake#L59-L63